diff -ruN ClanLib-0.6.5../Sources/Core/IOData/Unix/directory_scanner_unix.cpp ClanLib-0.6.5/Sources/Core/IOData/Unix/directory_scanner_unix.cpp
--- ClanLib-0.6.5../Sources/Core/IOData/Unix/directory_scanner_unix.cpp	2002-01-10 11:23:30.000000000 +0100
+++ ClanLib-0.6.5/Sources/Core/IOData/Unix/directory_scanner_unix.cpp	2003-06-03 22:57:48.000000000 +0200
@@ -13,6 +13,7 @@
         ------------------------------------------------------------------------
 */
 
+#include <cassert>
 #include <API/Core/System/error.h>
 #include <iostream>
 #include <libgen.h>
