--- ClanLib-0.7.4/Sources/Application/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.4/Sources/Application/Makefile.am	2003-10-09 23:15:08.843198880 +0200
@@ -6,6 +6,7 @@
 libclanApp_la_SOURCES = Unix/clanapp.cpp
 endif
 
+libclanApp_la_LIBADD = ../Core/libclanCore.la
 libclanApp_la_LDFLAGS = \
   -release $(LT_RELEASE) \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
--- ClanLib-0.7.8/Sources/Display/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.8/Sources/Display/Makefile.am	2003-10-09 23:16:41.208157272 +0200
@@ -3,7 +3,8 @@
 # FIXME: This needs to be @LIBS@ or @X11LIBS@ or something like that
 if WIN32
 else
-libclanDisplay_la_LIBADD = -L/usr/X11R6/lib/ -lX11 -lXmu -lGL -lGLU -lpng -ljpeg -lXxf86vm @X_LIBS@
+LXLIBDIR=-L/usr/X11R6/lib
+libclanDisplay_la_LIBADD = $(LXLIBDIR) -lX11 -lXmu -lGL -lGLU -lpng -ljpeg -lXxf86vm @X_LIBS@ ../Core/libclanCore.la
 endif
 
 libclanDisplay_la_SOURCES = \
--- ClanLib-0.8.0-RC1/Sources/GL/Makefile.am.orig	2005-02-11 15:13:43.000000000 +0100
+++ ClanLib-0.8.0-RC1/Sources/GL/Makefile.am	2006-01-25 23:38:11.091887750 +0100
@@ -51,7 +51,7 @@
 GLX/input_device_x11keyboard.h \
 GLX/input_device_xinput.cpp \
 GLX/input_device_xinput.h
-libclanGL_la_LIBADD = -lXxf86vm -lXi -lGLU -lGL 
+libclanGL_la_LIBADD = -lXxf86vm -lXi -lGLU -lGL ../Display/libclanDisplay.la
 endif
 
 EXTRA_DIST = \
--- ClanLib-0.7.4/Sources/GUI/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.4/Sources/GUI/Makefile.am	2003-10-09 23:19:13.152058288 +0200
@@ -82,6 +82,8 @@
 filedialog.cpp \
 window_generic.h
 
+libclanGUI_la_LIBADD = ../Display/libclanDisplay.la
+
 libclanGUI_la_LDFLAGS = \
   -release $(LT_RELEASE) \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
--- ClanLib-0.7.4/Sources/GUIStyleSilver/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.4/Sources/GUIStyleSilver/Makefile.am	2003-10-09 23:21:06.631806744 +0200
@@ -10,6 +10,7 @@
 image_silver.h     progressbar_silver.h  window_silver.h \
 inputbox_silver.h  radiobutton_silver.h
 
+libclanGUIStyleSilver_la_LIBADD = ../GUI/libclanGUI.la
 libclanGUIStyleSilver_la_LDFLAGS = \
   -release $(LT_RELEASE) \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
--- ClanLib-0.7.4/Sources/MikMod/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.4/Sources/MikMod/Makefile.am	2003-10-09 23:24:59.453412480 +0200
@@ -8,7 +8,7 @@
  soundprovider_mikmod.cpp   \
  soundprovider_mikmod_session.cpp
 
-libclanMikMod_la_LIBADD = -lmikmod
+libclanMikMod_la_LIBADD = -lmikmod ../Sound/libclanSound.la
 
 libclanMikMod_la_LDFLAGS = \
   -release $(LT_RELEASE) \
--- ClanLib-0.7.4/Sources/Network/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.4/Sources/Network/Makefile.am	2003-10-09 23:25:43.592702288 +0200
@@ -69,6 +69,7 @@
 libclanNetwork_la_SOURCES += setupnetwork_unix.cpp
 endif
 
+libclanNetwork_la_LIBADD = ../Core/libclanCore.la
 libclanNetwork_la_LDFLAGS = \
   -release $(LT_RELEASE) \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
--- ClanLib-0.7.4/Sources/SDL/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.4/Sources/SDL/Makefile.am	2003-10-09 23:26:23.644613472 +0200
@@ -20,7 +20,7 @@
 surface_sdl.h
 
 libclanSDL_la_CXXFLAGS = $(SDL_CFLAGS)
-libclanSDL_la_LIBADD   = $(SDL_LIBS)
+libclanSDL_la_LIBADD   = $(SDL_LIBS) ../Display/libclanDisplay.la
 
 libclanSDL_la_LDFLAGS = \
   -release $(LT_RELEASE) \
--- ClanLib-0.7.4/Sources/Sound/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.4/Sources/Sound/Makefile.am	2003-10-09 23:27:44.491322904 +0200
@@ -36,6 +36,7 @@
 Unix/soundoutput_oss.h
 endif
 
+libclanSound_la_LIBADD = ../Core/libclanCore.la
 libclanSound_la_LDFLAGS = \
   -release $(LT_RELEASE) \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
--- ClanLib-0.7.4/Sources/Vorbis/Makefile.am.orig	2003-09-15 16:18:50.000000000 +0200
+++ ClanLib-0.7.4/Sources/Vorbis/Makefile.am	2003-10-09 23:28:39.064026592 +0200
@@ -5,7 +5,7 @@
 	soundprovider_vorbis.cpp \
 	soundprovider_vorbis_session.cpp
 
-libclanVorbis_la_LIBADD = -logg -lvorbis
+libclanVorbis_la_LIBADD = -logg -lvorbis ../Sound/libclanSound.la
 
 libclanVorbis_la_LDFLAGS = \
   -release $(LT_RELEASE) \
--- ClanLib-0.8.0-RC1/Sources/Core/Makefile.am.orig	2005-07-10 01:25:31.000000000 +0200
+++ ClanLib-0.8.0-RC1/Sources/Core/Makefile.am	2006-01-26 21:57:59.647858500 +0100
@@ -125,7 +125,7 @@
 endif
 
 # FIXME: Hardcoding these here might not be a good idea
-libclanCore_la_LIBADD = @LIBPTHREAD@
+libclanCore_la_LIBADD = @LIBPTHREAD@ ../Signals/libclanSignals.la -ldl
 
 libclanCore_la_LDFLAGS = \
   -release $(LT_RELEASE) \
--- ClanLib-0.8.0-RC1/configure.ac.orig	2005-10-20 22:57:15.000000000 +0200
+++ ClanLib-0.8.0-RC1/configure.ac	2006-01-26 18:13:46.567093500 +0100
@@ -422,16 +422,15 @@
 CXXFLAGS="$CXXFLAGS -I\$(top_srcdir)/Sources"
 
 # Modules which will be build
-ClanLib_Modules="Application"
-ClanLib_pkgconfig="clanApp.pc"
+ClanLib_Modules="Signals Core Application"
+ClanLib_pkgconfig="clanSignals.pc clanCore.pc clanApp.pc"
 
 # API header files for the Modules which will be installed
-ClanLib_API_Modules="$ClanLib_API_Modules \$(clanApp_includes)"
+ClanLib_API_Modules="$ClanLib_API_Modules \$(clanSignals_includes) \$(clanCore_includes) \$(clanApp_includes)"
 
 # Default examples which will be build
 ClanLib_Examples="Directory Minimum Resources ResourcesCustom Signals ThreadSync XML Zip"
 
-CLANLIB_ENABLE_MODULES([Core Signals])
 
 if test "$enable_clanDisplay" = "yes"; then
     CLANLIB_ENABLE_MODULES(Display)
@@ -625,6 +624,8 @@
 AC_SUBST(ClanLib_Examples)
 
 AC_CONFIG_FILES([
+        pkgconfig/clanSignals.pc
+        pkgconfig/clanCore.pc
         pkgconfig/clanApp.pc
         pkgconfig/Makefile
         Makefile
@@ -632,6 +633,8 @@
         Examples/Makefile.conf
         Setup/Makefile
         Sources/API/Makefile
+        Sources/Signals/Makefile
+        Sources/Core/Makefile
         Sources/Application/Makefile
         Sources/Makefile
         Tests/Makefile.conf
