--- ClanLib-2.2.7/Examples/Makefile.conf.in~    2011-01-27 09:46:22.000000000 +0100
+++ ClanLib-2.2.7/Examples/Makefile.conf.in     2011-03-23 11:30:20.041337929 +0100
@@ -6,7 +6,7 @@

 PACKAGES = $(patsubst %,%-@LT_RELEASE@,$(LIBS))
 @WIN32_TRUE@ CXXFLAGS += `pkg-config --cflags $(PACKAGES)` -mthreads
-@WIN32_FALSE@CXXFLAGS += `pkg-config --cflags $(PACKAGES)` -pthread
+@WIN32_FALSE@CXXFLAGS += -I. -L../../../Sources/Core/.libs `pkg-config --cflags $(PACKAGES)` -pthread

 #EXAMPLE_BIN contains list of example binaries to build

