--- ClanLib-0.7.2/configure.in.orig	2003-02-25 22:46:52.000000000 +0100
+++ ClanLib-0.7.2/configure.in	2003-04-26 16:14:12.000000000 +0200
@@ -576,10 +576,9 @@
 AC_ARG_ENABLE(debug,[  --enable-debug          Enable debugging support],
 [
   if test "$enableval" == "yes"; then
-    CXXFLAGS="$CXXFLAGS -DDEBUG=1 -g3 -ggdb3 -O0"
+    CXXFLAGS="$CXXFLAGS -DDEBUG=1 -g3 -ggdb3"
     AC_MSG_RESULT([enabled])
   else
-    CXXFLAGS="$CXXFLAGS -O2"
     AC_MSG_RESULT([*********** <-- DISABLED --> ***********])
   fi
 ],
