diff -urN ClanLib-0.8.1/Sources/GL/canvas_opengl.cpp ClanLib-0.8.1.new/Sources/GL/canvas_opengl.cpp
--- ClanLib-0.8.1/Sources/GL/canvas_opengl.cpp	2007-10-21 02:14:01.000000000 +0200
+++ ClanLib-0.8.1.new/Sources/GL/canvas_opengl.cpp	2009-05-07 10:24:50.134346567 +0200
@@ -48,6 +48,7 @@
 #include "graphic_context_opengl.h"
 #include "surface_target_opengl.h"
 #include <cstring>
+#include <cstdio>
 
 /////////////////////////////////////////////////////////////////////////////
 // CL_Canvas_OpenGL construction:
diff -urN ClanLib-0.8.1/Sources/GL/GLX/input_device_linuxevent.cpp ClanLib-0.8.1.new/Sources/GL/GLX/input_device_linuxevent.cpp
--- ClanLib-0.8.1/Sources/GL/GLX/input_device_linuxevent.cpp	2007-10-21 02:14:01.000000000 +0200
+++ ClanLib-0.8.1.new/Sources/GL/GLX/input_device_linuxevent.cpp	2009-05-07 10:27:25.440582028 +0200
@@ -35,6 +35,7 @@
 #include <iostream>
 #include <errno.h>
 #include <cstring>
+#include <cstdio>
 
 #include "API/Core/System/error.h"
 #include "API/Core/System/clanstring.h"
diff -urN ClanLib-0.8.1/Sources/GL/GLX/input_device_xinput.cpp ClanLib-0.8.1.new/Sources/GL/GLX/input_device_xinput.cpp
--- ClanLib-0.8.1/Sources/GL/GLX/input_device_xinput.cpp	2007-10-21 02:14:01.000000000 +0200
+++ ClanLib-0.8.1.new/Sources/GL/GLX/input_device_xinput.cpp	2009-05-07 10:29:01.220572400 +0200
@@ -29,6 +29,7 @@
 
 #include <iostream>
 #include <cstring>
+#include <cstdio>
 #include "API/Core/System/log.h"
 #include "API/Display/input_device.h"
 #include "API/Display/input_event.h"
diff -urN ClanLib-0.8.1/Sources/GL/opengl.cpp ClanLib-0.8.1.new/Sources/GL/opengl.cpp
--- ClanLib-0.8.1/Sources/GL/opengl.cpp	2006-10-27 03:55:01.000000000 +0200
+++ ClanLib-0.8.1.new/Sources/GL/opengl.cpp	2009-05-07 10:25:50.247258791 +0200
@@ -27,6 +27,8 @@
 **    (if your name is missing here, please add it)
 */
 
+#include <cstring>
+
 #include "Display/display_precomp.h"
 #include "API/GL/opengl.h"
 #include "API/GL/opengl_wrap.h"
