--- ClanLib-2.2.7/Sources/Core/Makefile.am.orig	2010-07-19 09:10:06.000000000 +0200
+++ ClanLib-2.2.7/Sources/Core/Makefile.am	2011-03-26 14:43:21.808912129 +0100
@@ -164,7 +164,7 @@
 
 endif
 
-libclan30Core_la_LIBADD = @LIBPTHREAD@
+libclan30Core_la_LIBADD = @LIBPTHREAD@ -lpthread
 
 libclan30Core_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
--- ClanLib-2.2.7/Sources/Network/Makefile.am.orig	2011-01-27 09:46:22.000000000 +0100
+++ ClanLib-2.2.7/Sources/Network/Makefile.am	2011-03-26 14:55:17.676908777 +0100
@@ -45,9 +45,10 @@
 setupnetwork_unix.cpp
 endif
 
+libclan30Network_la_LIBADD = $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30Network_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
-  $(extra_LIBS_Network)
+  $(extra_LIBS_clanNetwork)
 
 libclan30Sound_la_CXXFLAGS=$(Network_CXXFLAGS) $(extra_CFLAGS_Network)
 
--- ClanLib-2.2.7/Sources/Sound/Makefile.am.orig	2010-07-05 16:50:31.000000000 +0200
+++ ClanLib-2.2.7/Sources/Sound/Makefile.am	2011-03-26 14:45:51.348911291 +0100
@@ -55,6 +55,7 @@
 SoundProviders/Unix/soundprovider_recorder_oss.h
 endif
 
+libclan30Sound_la_LIBADD = $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30Sound_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanSound)
--- ClanLib-2.2.7/Sources/CSSLayout/Makefile.am.orig	2010-07-05 16:50:31.000000000 +0200
+++ ClanLib-2.2.7/Sources/CSSLayout/Makefile.am	2011-03-26 14:47:45.184912129 +0100
@@ -380,6 +380,7 @@
 	precomp.cpp \
 	precomp.h
 
+libclan30CSSLayout_la_LIBADD = $(top_builddir)/Sources/Display/libclan30Display.la $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30CSSLayout_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanCSSLayout)
--- ClanLib-2.2.7/Sources/GL/Makefile.am.orig	2011-01-27 09:46:22.000000000 +0100
+++ ClanLib-2.2.7/Sources/GL/Makefile.am	2011-03-26 14:57:45.268909615 +0100
@@ -48,6 +48,7 @@
 	GLX/opengl_window_provider_glx.cpp \
 	GLX/opengl_window_provider_glx.h
 endif
+libclan30GL_la_LIBADD = $(top_builddir)/Sources/Display/libclan30Display.la $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30GL_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanGL)
--- ClanLib-2.2.7/Sources/Database/Makefile.am.orig	2010-11-01 09:26:08.000000000 +0100
+++ ClanLib-2.2.7/Sources/Database/Makefile.am	2011-03-26 15:00:40.788912129 +0100
@@ -8,6 +8,7 @@
 db_value.cpp \
 db_transaction.cpp
 
+libclan30Database_la_LIBADD = $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30Database_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanDatabase)
--- ClanLib-2.2.7/Sources/App/Makefile.am.orig	2010-07-05 16:50:31.000000000 +0200
+++ ClanLib-2.2.7/Sources/App/Makefile.am	2011-03-26 15:01:37.480908777 +0100
@@ -6,6 +6,7 @@
 libclan30App_la_SOURCES = Unix/clanapp.cpp
 endif
 
+libclan30App_la_LIBADD = $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30App_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanApp)
--- ClanLib-2.2.7/Sources/GUI/Makefile.am.orig	2010-07-05 16:50:31.000000000 +0200
+++ ClanLib-2.2.7/Sources/GUI/Makefile.am	2011-03-26 15:04:04.180909615 +0100
@@ -112,6 +112,7 @@
 gui_theme_default_impl.h \
 precomp.h
 
+libclan30GUI_la_LIBADD = $(top_builddir)/Sources/CSSLayout/libclan30CSSLayout.la $(top_builddir)/Sources/Display/libclan30Display.la $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30GUI_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanGUI)
--- ClanLib-2.2.7/Sources/Sqlite/Makefile.am.orig	2010-07-05 16:50:31.000000000 +0200
+++ ClanLib-2.2.7/Sources/Sqlite/Makefile.am	2011-03-26 15:06:37.852907938 +0100
@@ -9,9 +9,10 @@
 sqlite_reader_provider.cpp \
 sqlite_transaction_provider.cpp
 
+libclan30Sqlite_la_LIBADD = $(top_builddir)/Sources/Database/libclan30Database.la $(top_builddir)/Sources/Core/libclan30Core.la -lpthread -lsqlite3
 libclan30Sqlite_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
-  $(extra_LIBS_Sqlite)
+  $(extra_LIBS_clanSqlite)
 
 libclan30Sound_la_CXXFLAGS=$(Sqlite_CXXFLAGS) $(extra_CFLAGS_Sqlite)
 
--- ClanLib-2.3.1/Sources/Display/Makefile.am.orig      2011-07-26 11:43:05.000000000 +0200
+++ ClanLib-2.3.1/Sources/Display/Makefile.am   2011-08-10 09:48:40.939755410 +0200
@@ -1,6 +1,7 @@
 lib_LTLIBRARIES = libclan30Display.la

 if WIN32
+libclan30Display_la_LIBADD =
 else
 libclan30Display_la_LIBADD = -lpng -ljpeg
 endif
@@ -201,7 +202,7 @@
        X11/font_config.cpp \
        X11/font_config.h
 endif
-
+libclan30Display_la_LIBADD += $(top_builddir)/Sources/Core/libclan30Core.la $(clanDisplay_LIBS)
 libclan30Display_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanDisplay
--- ClanLib-2.2.7/Sources/SWRender/Makefile.am.orig	2010-07-05 16:50:31.000000000 +0200
+++ ClanLib-2.2.7/Sources/SWRender/Makefile.am	2011-03-27 07:33:24.821729730 +0200
@@ -69,9 +69,10 @@
 Canvas/Commands/pixel_command_triangle.cpp \
 swr_target.cpp
 
+libclan30SWRender_la_LIBADD = $(top_builddir)/Sources/Display/libclan30Display.la $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30SWRender_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
-  $(extra_LIBS_SWRender)
+  $(extra_LIBS_clanSWRender)
 
 libclan30Sound_la_CXXFLAGS=$(SWRender_CXXFLAGS) $(extra_CFLAGS_SWRender)
 
--- ClanLib-2.2.7/configure.ac.orig	2011-01-27 09:46:22.000000000 +0100
+++ ClanLib-2.2.7/configure.ac	2011-03-26 15:25:57.540910453 +0100
@@ -631,9 +631,14 @@
 dnl Set the include path
 CXXFLAGS="$CXXFLAGS -I\$(top_srcdir)/Sources"
 
+ClanLib_Modules=
+ClanLib_pkgconfig=
+ClanLib_API_Modules=
 dnl Modules which will be build
-ClanLib_Modules="App"
-ClanLib_pkgconfig="clanApp.pc"
+CLANLIB_ENABLE_MODULES(Core)
+
+ClanLib_Modules="$ClanLib_Modules App"
+ClanLib_pkgconfig="$ClanLib_pkgconfig clanApp.pc"
 
 dnl API header files for the Modules which will be installed
 ClanLib_API_Modules="$ClanLib_API_Modules \$(clanApp_includes)"
@@ -641,7 +646,6 @@
 dnl Default examples which will be build
 dnl (Currently Disabled)
 ClanLib_Examples=""
-CLANLIB_ENABLE_MODULES(Core)
 
 if test "$enable_clanDisplay" = "yes"; then
 	CLANLIB_ENABLE_MODULES(Display)
@@ -681,14 +686,14 @@
 if test "$enable_clanGL" = "yes"; then
 	CLANLIB_ENABLE_MODULES(GL)
 fi
+if test "$enable_clanCSSLayout" = "yes"; then
+	CLANLIB_ENABLE_MODULES(CSSLayout)
+fi
 
 if test "$enable_clanGUI" = "yes"; then
 	CLANLIB_ENABLE_MODULES(GUI)
 fi
 
-if test "$enable_clanCSSLayout" = "yes"; then
-	CLANLIB_ENABLE_MODULES(CSSLayout)
-fi
 if test "$enable_clanSWRender" = "yes"; then
 	CLANLIB_ENABLE_MODULES(SWRender)
 fi
--- ClanLib-3.0.0/Sources/Compute/Makefile.am.orig	2013-11-05 19:30:01.877180958 +0100
+++ ClanLib-3.0.0/Sources/Compute/Makefile.am	2013-11-05 19:30:17.203847516 +0100
@@ -21,6 +21,7 @@
 compute_program.cpp \
 setup_compute.cpp
 
+libclan30Compute_la_LIBADD = $(top_builddir)/Sources/Core/libclan30Core.la $(top_builddir)/Sources/GL/libclan30GL.la
 libclan30Compute_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_Compute)
--- ClanLib-3.0.0/Sources/Scene3D/Makefile.am.orig	2013-11-05 19:33:55.787179266 +0100
+++ ClanLib-3.0.0/Sources/Scene3D/Makefile.am	2013-11-05 19:34:39.483845615 +0100
@@ -46,6 +46,7 @@
 Performance/gpu_timer.cpp \
 Performance/scope_timer.cpp
 
+libclan30Scene3D_la_LIBADD = $(top_builddir)/Sources/Core/libclan30Core.la $(top_builddir)/Sources/GL/libclan30GL.la
 libclan30Scene3D_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanScene3D)
--- ClanLib-3.0.0/Sources/Physics3D/Makefile.am.orig	2013-11-05 19:35:36.817178534 +0100
+++ ClanLib-3.0.0/Sources/Physics3D/Makefile.am	2013-11-05 19:35:51.530511761 +0100
@@ -137,6 +137,7 @@
 physics3d_ray_test.cpp \
 physics3d_contact_test.cpp
 
+libclan30Physics3D_la_LIBADD = $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30Physics3D_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanPhysics3D)
--- ClanLib-3.0.0/Sources/Physics2D/Makefile.am.orig	2013-09-24 13:53:31.000000000 +0200
+++ ClanLib-3.0.0/Sources/Physics2D/Makefile.am	2013-11-05 19:37:26.527177740 +0100
@@ -82,6 +82,7 @@
 Collision/Shapes/circle_shape.cpp \
 Collision/physics_object.cpp
 
+libclan30Physics2D_la_LIBADD = $(top_builddir)/Sources/Display/libclan30Display.la $(top_builddir)/Sources/Core/libclan30Core.la
 libclan30Physics2D_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanPhysics2D)
--- ClanLib-3.0.0/Sources/GameIDE/Makefile.am.orig	2013-11-05 19:38:08.570510770 +0100
+++ ClanLib-3.0.0/Sources/GameIDE/Makefile.am	2013-11-05 19:39:03.023843708 +0100
@@ -157,6 +157,7 @@
 TextEditor/Scintilla/lexlib/LexerModule.cxx
 endif
 
+libclan30GameIDE_la_LIBADD = $(top_builddir)/Sources/GUI/libclan30GUI.la
 libclan30GameIDE_la_LDFLAGS = \
   -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LDFLAGS_LT_RELEASE) \
   $(extra_LIBS_clanGameIDE)
--- /home/users/baggins/devel/PLD/rpm/BUILD/ClanLib-3.0.0/Sources/Scene3D/Passes/GBuffer/gbuffer_pass.cpp~	2013-09-24 13:53:31.000000000 +0200
+++ /home/users/baggins/devel/PLD/rpm/BUILD/ClanLib-3.0.0/Sources/Scene3D/Passes/GBuffer/gbuffer_pass.cpp	2013-11-05 20:08:58.168398046 +0100
@@ -35,7 +35,7 @@
 
 namespace clan
 {
-
+extern cl_tls_variable GLFunctions *OpenGL::functions;
 
 GBufferPass::GBufferPass(ResourceContainer &inout)
 {
