--- ClanLib-2.1.0/Examples/Makefile.conf.in~	2009-04-28 14:42:43.000000000 +0000
+++ ClanLib-2.1.0/Examples/Makefile.conf.in	2009-10-21 12:03:34.000000000 +0000
@@ -5,7 +5,7 @@
 #
 
 PACKAGES = $(patsubst %,%-@LT_RELEASE@,$(LIBS))
-CXXFLAGS += `pkg-config --cflags $(PACKAGES)` -pthread
+CXXFLAGS += -I. -L../../../Sources/Core/.libs `pkg-config --cflags $(PACKAGES)` -pthread
 
 #EXAMPLE_BIN contains list of example binaries to build
 
--- ClanLib-2.1.0/Documentation/Reference/Makefile.am~	2009-03-27 10:25:47.000000000 +0000
+++ ClanLib-2.1.0/Documentation/Reference/Makefile.am	2009-10-21 12:04:46.000000000 +0000
@@ -11,7 +11,7 @@
 	mkdir reftest
 	doxygen ./clanlib.doxygen
 	make -C ../Utilities/ReferenceDocs
-	../Utilities/ReferenceDocs/ReferenceDocs
+	LD_LIBRARY_PATH=../../Sources/Core/.libs/ ../Utilities/ReferenceDocs/ReferenceDocs
 
 install-html:
 	$(INSTALL) -d $(DESTDIR)$(HTML_PREFIX)
